dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContainerLoader.java

```diff
@@ -331,31 +337,41 @@ private InputStream buildVelocity(final Container container, final String uuid,
                 velocityCodeBuilder.append("#if($HAVE_A_VERSION)");
                 
                     // ##Checking permission to see content
-                    if (mode.showLive) velocityCodeBuilder.append("#if($contents.doesUserHasPermission($CONTENT_INODE, 1, $user, true))");
+                    if (mode.showLive) {
+                        // flag `$_show_working_` is set to true only when time-machine is on
+                        //if time-machine's on.. no need to check for permissions since anonymous users can not visualize working content since 5.
+                        velocityCodeBuilder.append("#if($contents.doesUserHasPermission($CONTENT_INODE, 1, $user, true))");
+                    }
                     
                         // ### START BODY ###
                         velocityCodeBuilder.append("#if($isWidget==true)").append("$widgetCode");
                         velocityCodeBuilder.append("#elseif($isForm==true)").append("$formCode");
                         velocityCodeBuilder.append("#else");
-            
+
                             for (int i = 0; i < containerContentTypeList.size(); i++) {
                                 ContainerStructure cs = containerContentTypeList.get(i);
                                 String ifelse = (i == 0) ? "if" : "elseif";
                                 velocityCodeBuilder.append("#" + ifelse + "($ContentletStructure ==\"" + cs.getStructureId() + "\")");
                                 velocityCodeBuilder.append(cs.getCode());
                             }
-                            if (containerContentTypeList.size() > 0) velocityCodeBuilder.append("#end");
+                            if (containerContentTypeList.size() > 0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4504016)